File: 427A-PoliceRecruits/427a-policeRecruits.c
#include <stdio.h>
 
int main(int argc, char** argv) {
    int n, num;
    int readyToWork = 0, untreatedCrimes = 0;
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &num);
        if (num > 0) readyToWork += num;
        else {
            if (readyToWork > 0) readyToWork--;
            else untreatedCrimes++;
        }
    }
    printf("%d", untreatedCrimes);
}

File: 427A-PoliceRecruits/427a-policeRecruits.py
n = int(input())
tab = [int(x) for x in input().split()]
readyToWork = 0
untreatedCrimes = 0
for i in tab:
    if i > 0:
        readyToWork += i
    else:
        if readyToWork > 0:
            readyToWork -= 1
        else:
            untreatedCrimes += 1
print(untreatedCrimes)
